src/include/distributed/pg_dist_partition.h

```diff
@@ -23,6 +23,7 @@ typedef struct FormData_pg_dist_partition
 {
 	Oid logicalrelid;              /* logical relation id; references pg_class oid */
 	char partmethod;               /* partition method; see codes below */
+	int colocationid;
```

Comment:
Yes, it's a problem. To solve it I'd suggest simply putting the field at the end, and using the normal tuple accessors /  tuple deforming.